tests/ga/test_extension.py

```diff
@@ -2535,304 +2536,285 @@ def test_ext_handler_with_systemd_cgroup_enabled(self, *args):
-        error_test_file = """
-                    exit %s
-                    """ % exit_code
-
-        test_env_file = """
-            printenv | grep AZURE_
-            """
-
-        self.create_script(file_name=test_env_file_name, contents=test_env_file,
-                           file_path=os.path.join(new_handler_i.get_base_dir(), test_env_file_name))
-        self.create_script(file_name=test_failure_file_name, contents=error_test_file,
-                           file_path=os.path.join(old_handler_i.get_base_dir(), test_failure_file_name))
-
-        with patch.object(new_handler_i, 'report_event', autospec=True) as mock_report:
-
-            uninstall_rc = ExtHandlersHandler._update_extension_handler_and_return_if_failed(old_handler_i, new_handler_i)
-            _, kwargs = mock_report.call_args
-
-            self.assertEqual(exit_code, uninstall_rc)
-            self.assertIn("%s=%s" % (ExtCommandEnvVariable.DisableReturnCode, exit_code), kwargs['message'])
-
-    @patch('time.sleep', side_effect=lambda _: mock_sleep(0.0001))
-    def test_timeout_code_should_set_on_cmd_timeout(self, _):
-        test_env_file_name = "test_env.sh"
-        test_failure_file_name = "test_fail.sh"
-        old_handler_i = TestExtensionUpdateOnFailure._get_ext_handler_instance('foo', '1.0.0', handler={
-            "disableCommand": test_failure_file_name,
-            "uninstallCommand": test_failure_file_name})
-        new_handler_i = TestExtensionUpdateOnFailure._get_ext_handler_instance(
-            'foo', '1.0.1',
-            handler={"updateCommand": test_env_file_name + " -u", "installCommand": test_env_file_name + " -i"},
-            continue_on_update_failure=True
-        )
-
-        exit_code = 156
-        error_test_file = """
-            sleep 1m
-            exit %s
-        """ % exit_code
+        fail_disable = lambda *args, **kwargs: exit_codes["disable"]
+        fail_uninstall = lambda *args, **kwargs: exit_codes["uninstall"]
 
-        test_env_file = """
-            printenv | grep AZURE_
-        """
+        first_ext = get_extension_actor(disableAction=fail_disable, uninstallAction=fail_uninstall)
+        second_ext = get_extension_actor(version="1.1.0", continueOnUpdateFailure=True)
+
+        TestExtensionUpdateOnFailure._do_upgrade_scenario(first_ext, second_ext)
+
+        first_ext.get_command("disableCommand").assert_called_once()
+        second_ext.get_command("updateCommand").assert_called_once()
+        first_ext.get_command("uninstallCommand").assert_called_once()
+        second_ext.get_command("installCommand").assert_called_once()
+
+        _, update_kwargs = second_ext.get_command("updateCommand").call_args
+        _, install_kwargs = second_ext.get_command("installCommand").call_args
+
+        second_extension_dir = "/".join([
+            conf.get_lib_dir(),
+            Formats.FormatExtensionDir(second_ext.extension_info.name, second_ext.extension_info.version)
+        ])
```

Comment:
Would it be cleaner to move this utility to the extension actor's class? I saw you used it several times in the tests.

Reply:
`Formats.FormatExtensionDir` is actually from that file; it might be best to simply bring this functionality into that method.